src/main/kotlin/no/nav/familie/ks/sak/kjerne/avstemming/GrensesnittavstemmingScheduler.kt

```diff
@@ -0,0 +1,58 @@
+package no.nav.familie.ks.sak.kjerne.avstemming
+
+import no.nav.familie.kontrakter.felles.objectMapper
+import no.nav.familie.ks.sak.kjerne.avstemming.domene.GrensesnittavstemmingTaskDto
+import no.nav.familie.ks.sak.task.nesteGyldigeTriggertidForBehandlingIHverdager
+import no.nav.familie.leader.LeaderClient
+import no.nav.familie.prosessering.domene.Status
+import no.nav.familie.prosessering.domene.Task
+import no.nav.familie.prosessering.internal.TaskService
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.data.domain.Pageable
+import org.springframework.scheduling.annotation.Scheduled
+import org.springframework.stereotype.Service
+import java.time.LocalDate
+
+@Service
+class GrensesnittavstemmingScheduler(private val taskService: TaskService) {
+
+    @Scheduled(cron = "\${CRON_GRENSESNITT_AVSTEMMING}")
+    fun utfør() {
+        if (LeaderClient.isLeader() != true) {
+            return
+        }
+        logger.info("Starter GrensesnittavstemmingScheduler..")
+        logger.info("Finner siste kjørte ${GrensesnittavstemmingTask.TASK_STEP_TYPE}")
+        val alleFerdigeGrensesnittavstemmingTasker = taskService.finnTasksMedStatus(
+            status = listOf(Status.FERDIG),
+            type = GrensesnittavstemmingTask.TASK_STEP_TYPE,
+            page = Pageable.unpaged()
+        )
+        // intielt fom, tom når det ikke finnes noen tasker
+        var fom = LocalDate.now().minusDays(1).atStartOfDay()
+        var tom = LocalDate.now().atStartOfDay()
+
+        // setter fom og tom basert på task data fra siste kjørte task
+        if (alleFerdigeGrensesnittavstemmingTasker.isNotEmpty()) {
+            val sisteFerdigTask = alleFerdigeGrensesnittavstemmingTasker.maxBy { it.opprettetTid }
+            logger.info("Fant siste ferdig task med id=${sisteFerdigTask.id}")
+            val sisteFerdigTaskData =
+                objectMapper.readValue(sisteFerdigTask.payload, GrensesnittavstemmingTaskDto::class.java)
+            fom = sisteFerdigTaskData.tom.toLocalDate().atStartOfDay()
+            tom = nesteGyldigeTriggertidForBehandlingIHverdager((24 * 60).toLong(), fom).toLocalDate().atStartOfDay()
```

Comment:
bør ikke tom være samme som triggertid for task?

Reply:
task trenger ikke en triggerTid nå, siden scheduler oppretter tasken

Reply:
men kan ikke datoene havne i usynk med taskene siden `nesteGyldigeTriggertidForBehandlingIHverdager ` tar høyde for faste helligdager?
feks. så lager scheduleren en task dagen før 1. mai da vil neste tom bli 2. mai. når scheduleren kjører 1. mai og oppretter en ny task vil den ta fom dato fra sist kjørte task som  er i fremtiden vil den ikke?